src/primitives/block.cpp

```diff
@@ -25,9 +25,8 @@ std::string CBlock::ToString() const
         hashMerkleRoot.ToString(),
         nTime, nBits, nNonce,
         vtx.size());
-    for (unsigned int i = 0; i < vtx.size(); i++)
-    {
-        s << "  " << vtx[i]->ToString() << "\n";
+    for (const auto &tx : vtx) {
```

Comment:
I did check, `auto&` is much more common in the codebase (60 times vs. sth. like 5 times).

> It seems counter intuitive to me since it is the variable that is a reference, not the type.

It's actually the type. It's apparent if you look at function signatures.

`void foo(const std::string&);` is a valid function signature (first argument is a reference type).